code/weapon/weapons.cpp

```diff
@@ -8020,31 +8038,49 @@ void weapon_spew_stats(WeaponSpewType type)
 	mprintf(("\n"));
 	for (auto &wi : Weapon_info)
 	{
-		if (wi.subtype != WP_LASER)
-			continue;
-		if (wi.wi_flags[Weapon::Info_Flags::Beam])
+		if (wi.subtype != WP_LASER && wi.subtype != WP_BEAM)
 			continue;
 
 		if (all_weapons || wi.wi_flags[Weapon::Info_Flags::Player_allowed])
 		{
 			mprintf(("%s\n", wi.name));
-			mprintf(("\tVelocity: %-11.0fRange: %.0f\n", wi.max_speed, wi.max_speed * wi.lifetime));
+			//Beam range is set in the b_info and velocity isn't very relevant to them.
+			if (wi.wi_flags[Weapon::Info_Flags::Beam])
+				mprintf(("\tVelocity: N/A Range: %.0f\n", wi.b_info.range));
+			else
+				mprintf(("\tVelocity: %-11.0fRange: %.0f\n", wi.max_speed, wi.max_speed* wi.lifetime));
```

Comment:
Note the `%-11.0f` marker which establishes the spacing.  The table line (for Subach) looks like this:
```
Subach HL-7
	Velocity: 450        Range: 900
```

For the N/A line you will need to align the spacing, so do this:
```
"\tVelocity: N/A        Range: %.0f\n"
```

Reply:
done. sample output:
```Vishnan Proximity Gun
	Velocity: 1000       Range: 1800
	DPS: 120 Hull, 120 Shield, 24 Subsystem
	Power Use: 1.2W      ROF: 4/s

Vishnan Gravity Spike
	Velocity: N/A        Range: 700
	DPS: 78 Hull, 78 Shield, 194 Subsystem
	Power Use: 2.5W      ROF: 1/s```